### PR TITLE
test: api_runtime_test: add new checkpoints

### PR DESCRIPTION
**Pull request by chuanchang**
Add new checkpoints into runtime testing.

Signed-off-by: Alex Jia <chuanchang.jia@gmail.com>

### Ⅰ. Describe what this PR did
As summary.

### Ⅳ. Describe how to verify it
go test

### CODE REVIEW

File: apis/opts/config/runtime.go [hunk 1]
```diff
@@ -50,6 +51,8 @@ func (r *Runtime) String() string {
 		str = append(str, fmt.Sprintf("%s", k))
 	}
 
+	// make sure string list is sorted
+	sort.Strings(str)
```
[the remark this answers is not in the document]
Comment (Ace-Tang) [reply apis/opts/config/runtime.go]: Agree,do not need slice for sort.

[the remark this answers is not in the document]
Comment (chuanchang) [reply apis/opts/config/runtime.go]: @fuweid @Ace-Tang the similar question to PR2328

**Comment by chuanchang** [comment]
@Ace-Tang @fuweid I have removed String() related testing according to zhijin's advise.

**Comment by zhuangqh** [comment]
missing the test for `Type()` interface? @chuanchang

**Comment by chuanchang** [comment]
> missing the test for `Type()` interface? @chuanchang

Done

**Pull request merged**